zanata-war/src/main/java/org/zanata/action/MergeTransAction.java

```diff
@@ -0,0 +1,230 @@
+
+    @Getter
+    @Setter
+    private String targetVersionSlug;
+
+    @Getter
+    private String sourceProjectSlug;
+
+    @Getter
+    @Setter
+    private String sourceVersionSlug;
+
+    @Getter
+    @Setter
+    private boolean keepExistingTranslation;
+
+    @In
+    private ProjectDAO projectDAO;
+
+    @In
+    private ProjectIterationDAO projectIterationDAO;
+
+    @In
+    private MergeTranslationsManager mergeTranslationsManager;
+
+    @In
+    private CopyTransManager copyTransManager;
+
+    @In
+    private CopyVersionManager copyVersionManager;
+
+    @In
+    private Messages msgs;
+
+    private HProjectIteration targetVersion;
+
+    private HProject sourceProject;
+
+    public void setTargetProjectSlug(String targetProjectSlug) {
+        this.targetProjectSlug = targetProjectSlug;
+
+        /**
+         * This should only true during first instantiation. See pages.xml
+         * for setter of mergeTransAction.targetProjectSlug
+         */
+        if(sourceProjectSlug == null) {
+            setSourceProjectSlug(targetProjectSlug);
+        }
+    }
+
+    public void setSourceProjectSlug(String sourceProjectSlug) {
+        if(!StringUtils.equals(this.sourceProjectSlug, sourceProjectSlug)) {
+            this.sourceProjectSlug = sourceProjectSlug;
+            sourceProject = null;
+            if (!getSourceProject().getProjectIterations().isEmpty()) {
+                sourceVersionSlug =
+                        getSourceProject().getProjectIterations().get(0)
+                                .getSlug();
+            }
+        }
+    }
+
+    public HProjectIteration getTargetVersion() {
+        if(targetVersion == null) {
+            targetVersion =
+                    projectIterationDAO.getBySlug(targetProjectSlug,
+                            targetVersionSlug);
+        }
+        return targetVersion;
+    }
+
+    public HProject getSourceProject() {
+        if(sourceProject == null) {
+            sourceProject = projectDAO.getBySlug(sourceProjectSlug);
+        }
+        return sourceProject;
+    }
+
+    public List<HProjectIteration> getSourceVersions() {
+        List<HProjectIteration> versions =
+                getSourceProject().getProjectIterations();
+
+        if(versions.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        //remove target version if both are the same project or obsolete
+        if (StringUtils.equals(sourceProjectSlug, targetProjectSlug)) {
+            for (HProjectIteration version : versions) {
+                if (version.getSlug().equals(targetVersionSlug)
+                        || version.getStatus().equals(EntityStatus.OBSOLETE)) {
+                    versions.remove(version);
+                    break;
+                }
+            }
+        }
+        return versions;
+    }
+
+    public List<HProject> getProjects() {
+        return projectDAO.getProjects(EntityStatus.ACTIVE,
+                EntityStatus.READONLY);
+    }
```

Comment:
Could filter the projects here to only show projects that have 1 or more non-obsolete versions. This might be too surprising to users though, so maybe it is better just to leave it as-is.

@lukebrooker any thoughts?